### PR TITLE
misc: Docs Update + Other non-functional changes

### PR DESCRIPTION
**Pull request by tzaffi**
I pulled out the doc updates from draft PR #83 and also fixed a couple of scoping issues in our tests.

## Summary of Changes

* Introducing `.markdownlint.yml`. This isn't enforced in CI but while using the VS Code extension it helped me identify some bugs in our docs. For example, some [improperly formatted list items](https://developer.algorand.org/docs/get-details/conduit/Development/#creating-a-plugin).
*  Small edits in the following docs:
   * `docs/Configuration.md`
   * `docs/Development.md`
   * `docs/GettingStarted.md`
   * `docs/tutorials/WritingBlocksToFile.md`
* loop variable capture fixes in:
   * `conduit/pipeline/pipeline_test.go`
   * `conduit/plugins/importers/algod/algod_importer_test.go` (also tightening the test expectations around the genesis object received from `Init()`)

## TODO

- [x] Shall we add `exportloopref` to our linters?

```.golangci.yml
linters:
  enable:
    - exportloopref
```

### CODE REVIEW

File: conduit/pipeline/pipeline_test.go [hunk 1]
```diff
@@ -612,6 +612,7 @@ func TestRoundRequestError(t *testing.T) {
 func TestRoundOverride(t *testing.T) {
 	// cli override NextRound, 0 is a test for no override.
 	for i := 0; i < 10; i++ {
+		i := i
```
Comment (algochoi) [inline conduit/pipeline/pipeline_test.go:615]: good catch

File: .markdownlint.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+default: true
```
Comment (algochoi) [inline .markdownlint.yml:1]: I personally use VSCode so I am fine with this change but wondering what others think about adding IDE/editor/extension specific config files in the root dir.

**Review by algochoi — APPROVED** [review]
Looks like good changes to me!

File: .markdownlint.yml (hunk 2 above)
Comment (tzaffi) [reply .markdownlint.yml:1]: What do you think about adding `exportloopref` ?

File: conduit/pipeline/pipeline_test.go (hunk 1 above)
Comment (tzaffi) [reply conduit/pipeline/pipeline_test.go:615]: see cbe1692504021f71c69af1b636aa00cbe6015285

FYI - it didn't actually catch the problem. But we're also using it in `go-algorand` so it should be safe.

File: .markdownlint.yml (hunk 2 above)
Comment (tzaffi) [reply .markdownlint.yml:1]: see cbe1692504021f71c69af1b636aa00cbe6015285

FYI - it didn't actually catch the problem. But we're also using it in `go-algorand` so it should be safe.

**Review by winder — APPROVED** [review]

**Pull request merged**